src/main/java/org/prebid/server/bidder/readpeak/ReadPeakBidder.java

```diff
@@ -0,0 +1,224 @@
+
+        final BidRequest modifiedRequest = request.toBuilder().imp(modifiedImps).build();
+        final HttpRequest<BidRequest> httpRequest = makeHttpRequest(modifiedRequest, extImpReadPeaks.get(0));
+
+        return Result.of(Collections.singletonList(httpRequest), errors);
+    }
+
+    private ExtImpReadPeak parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), READPEAK_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Failed to deserialize ReadPeak extension: " + e.getMessage());
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpReadPeak extImpReadPeak) {
+        return imp.toBuilder()
+                .bidfloor(extImpReadPeak.getBidFloor() != null ? extImpReadPeak.getBidFloor() : imp.getBidfloor())
+                .tagid(StringUtils.isNotBlank(extImpReadPeak.getTagId()) ? extImpReadPeak.getTagId() : imp.getTagid())
+                .build();
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest request, ExtImpReadPeak extImpReadPeak) {
+        final Publisher publisher = Publisher.builder()
+                .id(extImpReadPeak.getPublisherId())
+                .build();
+
+        final BidRequest.BidRequestBuilder requestBuilder = request.toBuilder();
+
+        final boolean hasSite = request.getSite() != null;
+        final boolean hasApp = !hasSite && request.getApp() != null;
+
+        final BidRequest outgoingRequest = requestBuilder
+                .site(hasSite ? modifySite(request.getSite(), extImpReadPeak, publisher) : request.getSite())
+                .app(hasApp ? modifyApp(request.getApp(), extImpReadPeak, publisher) : request.getApp())
+                .build();
+
+        return BidderUtil.defaultRequest(outgoingRequest, endpointUrl, mapper);
+    }
+
+    private Site modifySite(Site site, ExtImpReadPeak extImpReadPeak, Publisher publisher) {
+        return site.toBuilder()
+                .id(StringUtils.isNotBlank(extImpReadPeak.getSiteId())
+                        ? extImpReadPeak.getSiteId() : site.getId())
+                .publisher(publisher)
+                .build();
+    }
+
+    private App modifyApp(App app, ExtImpReadPeak extImpReadPeak, Publisher publisher) {
+        return app.toBuilder()
+                .id(StringUtils.isNotBlank(extImpReadPeak.getSiteId())
+                        ? extImpReadPeak.getSiteId() : app.getId())
+                .publisher(publisher)
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return extractBids(bidResponse);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
```

Comment:
please use an inspiration `org.prebid.server.bidder.adview.AdviewBidder#makeBids` to collect errors

Reply:
I meant the following part
```java
       bidResponse.getSeatbid().stream()
                .filter(Objects::nonNull)
                .map(SeatBid::getBid)
                .filter(Objects::nonNull)
                .flatMap(Collection::stream)
                .map(bid -> makeBid(bid, bidResponse.getCur(), errors))
                .filter(Objects::nonNull)
                .toList();
```